### ISSUE TITLE
Dashboard stats wrongly shows zeroes

### ISSUE THREAD

**Issue by nbradbury** [woocommerce/woocommerce-android#861]
User (@kyleaparker) reports their dashboard stats showing zeroes when they should be showing actual data. Her stats should be showing 17 orders in this period. Note that her stats do show up correctly on iOS.

Looking at the app log I see a few instances of this FluxC crash, which may or may not be related:

```
java.lang.IllegalArgumentException: The input currency code is not 3-letter alphabetic code.
    at android.icu.util.Currency.getInstance(Currency.java:303)
    at java.util.Currency.getInstance(Currency.java:96)
    at org.wordpress.android.fluxc.utils.WCCurrencyUtils.getLocalizedCurrencySymbolForCode(WCCurrencyUtils.kt:45)
    at org.wordpress.android.fluxc.store.WooCommerceStore.formatCurrencyForDisplay(WooCommerceStore.kt:141)
    at com.woocommerce.android.util.CurrencyFormatter.formatCurrencyRounded(CurrencyFormatter.kt:64)
    at com.woocommerce.android.ui.dashboard.DashboardFragment$onActivityCreated$1.invoke(DashboardFragment.kt:97)
```
![53525414-b745bf00-3aaf-11e9-914f-462d4681cdc2](https://user-images.githubusercontent.com/3903757/53568247-74c3c700-3b2f-11e9-957e-ba2afa69a4e2.jpeg)

**Comment by rachelmcr** [comment]
We have two reports of this from external beta testers), and I'm seeing the same issue as well on my end (all on 1.3-rc-1). Let me know if you'd like to gather any additional details from me or the other beta testers to investigate this.

**Comment by nbradbury** [comment]
@rachelmcr Could you ask one of the testers to open the Dashboard, then send you their application log? I'm wondering if their log will show the same problem as Kylea's.

**Comment by nbradbury** [comment]
Looks like the invalid currency code isn't the problem. I was able to repro this on API 21.

![screenshot_1551360742](https://user-images.githubusercontent.com/3903757/53570032-83ac7880-3b33-11e9-865a-51957be97317.png)

**Comment by nbradbury** [comment]
Okay, all the above are red herrings. When running in the debugger I noticed that the stats retrieved from the DB are empty, then I noticed the below exception in the log.

It's not uncommon for devs to experience this sort of problem because we switch branches so often, but I'm wondering if we shipped a release with a faulty DB migration? If that's the case, then clearing the app's storage, or uninstalling and reinstalling, will fix the problem.

```
    android.database.sqlite.SQLiteException: table WCOrderStatsModel has no column named END_DATE (code 1): , while compiling: INSERT INTO WCOrderStatsModel(END_DATE,START_DATE,UNIT,QUANTITY,DATE,FIELDS,DATA,IS_CUSTOM_FIELD,LOCAL_SITE_ID) VALUES (?,?,?,?,?,?,?,?,?)
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:889)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:500)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
        at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
        at android.database.sqlite.SQLiteDatabase.insertWithOnConflict(SQLiteDatabase.java:1469)
        at android.database.sqlite.SQLiteDatabase.insert(SQLiteDatabase.java:1341)
        at com.yarolegovich.wellsql.InsertQuery.execute(InsertQuery.java:61)
        at org.wordpress.android.fluxc.persistence.WCStatsSqlUtils.insertOrUpdateStats(WCStatsSqlUtils.kt:32)
```

**Comment by nbradbury** [comment]
I see what the problem is, and it looks like @AmandaRiu is already on top of this with [this PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1141).

But for reference, the problem is that `WCOrderStatsModelTable` has this:

```
public static final String START_DATE = "START_DATE";
public static final String END_DATE = "END_DATE";
```

But in our migration to db v54 in `WellSqlConfig` we have this:

```
db.execSQL("ALTER TABLE WCOrderStatsModel ADD ENDDATE TEXT");
db.execSQL("ALTER TABLE WCOrderStatsModel ADD STARTDATE TEXT");
```

The difference in field names causes the stats data to not be stored.

**Comment by rachelmcr** [comment]
Just to confirm, I'm not going to reach out to testers at this point because it sounds like you've got the issue covered. (On my end, I'm on API 27 and don't have the currency error in my logs; the currency symbol shows up fine although the actual revenue/order numbers and graph don't load. So the symbol not showing up looks like a separate bug from the DB issue.)

**Comment by nbradbury** [comment]
I'll leave this open until the FluxC fix has been merged and we've updated the hash in WCAndroid.

**Comment by nbradbury** [comment]
Fixed via #869
